api/lql.go

```diff
@@ -19,151 +19,44 @@
-	// translate
-	if err := q.Translate(); err != nil {
-		return err
-	}
-	// validate range
-	if err := q.ValidateRange(allowEmptyTimes); err != nil {
-		return err
-	}
-	// validate query
-	if q.QueryText == "" {
-		return errors.New("query should not be empty")
-	}
-	return nil
+type UpdateQuery struct {
+	QueryText string `json:"queryText"`
 }
 
-func (q *LQLQuery) Translate() error {
-	// query
-	if err := q.TranslateQuery(); err != nil {
-		return err
-	}
-	// start
-	start, err := q.TranslateTime(q.StartTimeRange)
-	if err != nil {
-		return err
-	}
-	q.StartTimeRange = start
-	// end
-	end, err := q.TranslateTime(q.EndTimeRange)
-	if err != nil {
-		return nil
-	}
-	q.EndTimeRange = end
-
-	return nil
+type Query struct {
+	QueryID        string                   `json:"queryId"`
+	QueryText      string                   `json:"queryText"`
+	EvaluatorID    string                   `json:"evaluatorId"`
+	Owner          string                   `json:"owner"`
+	LastUpdateTime string                   `json:"lastUpdateTime"`
+	LastUpdateUser string                   `json:"lastUpdateUser"`
+	ResultSchema   []map[string]interface{} `json:"resultSchema"`
 }
 
-func (q *LQLQuery) TranslateQuery() error {
-	// if query text is already populated
-	if q.QueryText != "" {
-		return nil
-	}
-	// valid json
-	var t LQLQuery
-	if err := json.Unmarshal([]byte(q.QueryBlob), &t); err == nil {
-		if q.StartTimeRange == "" {
-			q.StartTimeRange = t.StartTimeRange
-		}
-		if q.EndTimeRange == "" {
-			q.EndTimeRange = t.EndTimeRange
-		}
-		q.QueryText = t.QueryText
-		return err
-	}
-	// invalid json
-	qblob := strings.ToLower(q.QueryBlob)
-	if strings.Contains(qblob, "start_time_range") ||
-		strings.Contains(qblob, "end_time_range") ||
-		strings.Contains(qblob, "lql_id") ||
-		strings.Contains(qblob, "query_text") {
-
-		return errors.New(LQLQueryTranslateError)
-	}
-	// valid lql text
-	if strings.Contains(q.QueryBlob, "{") &&
-		strings.Contains(q.QueryBlob, "}") {
-		q.QueryText = q.QueryBlob
-
-		return nil
-	}
-	// invalid lql text
-	return errors.New(LQLQueryTranslateError)
+type QueryResponse struct {
+	Data    Query  `json:"data"`
+	Message string `json:"message"`
 }
 
-func (q LQLQuery) TranslateTime(inTime string) (string, error) {
-	// empty
-	if inTime == "" {
-		return "", nil
-	}
-	// parse time as relative
-	if t, err := lwtime.ParseRelative(inTime); err == nil {
-		return t.UTC().Format(time.RFC3339), err
-	}
-	// parse time as RFC3339
-	if t, err := time.Parse(time.RFC3339, inTime); err == nil {
-		return t.UTC().Format(time.RFC3339), err
-	}
-	// parse time as millis
-	if i, err := strconv.ParseInt(inTime, 10, 64); err == nil {
-		return time.Unix(0, i*int64(time.Millisecond)).UTC().Format(time.RFC3339), err
-	}
-	return "", errors.New(fmt.Sprintf("unable to parse time (%s)", inTime))
+type QueriesResponse struct {
+	Data    []Query `json:"data"`
+	Message string  `json:"message"`
```

Comment:
What is this `Message`field? I don't see it in the API v2 specifications

Reply:
Message comes out when error.
```
❯ curl --location --request GET 'https://qan.qan.corp.lacework.net/api/v2/Queries/MyExampleQuery' \
--header 'Authorization: Bearer _ee0d0a838d01a4e77cb3170791810722'
{"message":"Error: Query id MyExampleQuery not found."}
```

Reply:
@afiune , I'm going to merge as-is with `Message` in the Response structs and I'll visit this with you again on the next PR...